ui/src/app/store/machine/slice.ts

```diff
@@ -359,867 +360,1338 @@ const statusHandlers = generateStatusHandlers<
       method: "action",
       statusKey: action.status,
-      prepare: (systemId: Machine[MachineMeta.PK]) => ({
-        action: action.name,
-        extra: {},
-        system_id: systemId,
-      }),
     };
-    switch (action.name) {
-      case "apply-storage-layout":
-        handler.method = "apply_storage_layout";
-        handler.prepare = (params: {
-          systemId: Machine[MachineMeta.PK];
-          storageLayout: StorageLayout;
-        }) => ({
-          storage_layout: params.storageLayout,
-          system_id: params.systemId,
-        });
-        break;
-      case "check-power":
-        handler.method = "check_power";
-        handler.prepare = (systemId: Machine[MachineMeta.PK]) => ({
-          system_id: systemId,
-        });
-        break;
-      case NodeActions.COMMISSION:
-        handler.prepare = ({
-          systemId,
-          enableSSH,
-          skipBMCConfig,
-          skipNetworking,
-          skipStorage,
-          updateFirmware,
-          configureHBA,
-          commissioningScripts,
-          testingScripts,
-          scriptInputs,
-        }: {
-          systemId: Machine[MachineMeta.PK];
-          enableSSH: boolean;
-          skipBMCConfig: boolean;
-          skipNetworking: boolean;
-          skipStorage: boolean;
-          updateFirmware: boolean;
-          configureHBA: boolean;
-          commissioningScripts: Script[];
-          testingScripts: Script[];
-          scriptInputs: ScriptInput[];
-        }) => {
-          let formattedCommissioningScripts: (string | Script["id"])[] = [];
-          if (commissioningScripts && commissioningScripts.length > 0) {
-            formattedCommissioningScripts = commissioningScripts.map(
-              (script) => script.id
-            );
-            if (updateFirmware) {
-              formattedCommissioningScripts.push("update_firmware");
-            }
-            if (configureHBA) {
-              formattedCommissioningScripts.push("configure_hba");
-            }
+    return handler;
+  }),
+  setErrors
+);
+
+const machineSlice = createSlice({
+  name: MachineMeta.MODEL,
+  initialState: {
+    ...genericInitialState,
+    active: null,
+    eventErrors: [],
+    selected: [],
+    statuses: {},
+  } as MachineState,
+  reducers: {
+    ...generateCommonReducers<
+      MachineState,
+      MachineMeta.PK,
+      CreateParams,
+      UpdateParams
+    >(MachineMeta.MODEL, MachineMeta.PK, setErrors),
+    [NodeActions.ABORT]: {
+      prepare: (system_id: Machine[MachineMeta.PK]) => ({
+        meta: {
+          model: MachineMeta.MODEL,
+          method: "action",
+        },
+        payload: {
+          params: {
+            action: NodeActions.ABORT,
+            extra: {},
+            system_id,
+          },
+        },
+      }),
+      reducer: () => {
+        // No state changes need to be handled for this action.
+      },
+    },
+    abortStart: statusHandlers.abort.start,
```

Comment:
Should we be using the `NodeActions` enum for these as well?
``` typescript
[`${NodeActions.ABORT}Start`]: statusHandlers[NodeActions.ABORT].start,
```

Reply:
Done.